src/Sylius/Bundle/PayumBundle/Storage/DoctrineStorage.php

```diff
@@ -34,11 +35,13 @@ public function __construct(protected ObjectManager $objectManager, $modelClass)
 
     public function findBy(array $criteria): array
     {
+        Assert::classExists($this->modelClass);
```

Comment:
1. Why not do it at constructor?
2. Is `* @param class-string $modelClass` not neccesary over here?
3. Why not add `string $modelClass` type-hint?

Reply:
The error is:
`Parameter #1 $className of method Doctrine\Persistence\ObjectManager::getRepository() expects class-string<object>, string given.`

It is not sufficient to assert that class exists in constructor, because a property can change in any moment. The only other way to fix is to override the property declaration on the class like this:
```php
    /**
     * @var class-string
     */
    protected $modelClass;
```
But IMHO is better to not override anything, if possible.. What do you think?

Reply:
It's an internal class, so we don't have to bother too much about it. I suggest to override the property.